fe/fe-core/src/main/java/com/starrocks/common/FeConstants.java

```diff
@@ -64,4 +64,6 @@
     public static String null_string = "\\N";
 
     public static boolean USE_MOCK_DICT_MANAGER = false;
+
+    public static final int default_tablet_number = 128;
```

Comment:
Could you explain why we choose `128` except `256` or `64`?

Reply:
Quite hard. It's just a rule-of-thumb. But the reason not to set this value too low is


- The cost of broadcast join, is to `parallelExecInstanceNum* beNum * outputSize`
- `parallelExecInstanceNum` is determined by `getLeftMostScanTabletsNum`(which is tablets number the scan operator scans)
- In most case, the `parallelExecInstanceNum` is actually `ConnectContext.get().getSessionVariable().getDegreeOfParallelism()`, so we want to set tablet number high enough.

```
        @Override
        public CostEstimate visitPhysicalDistribution(PhysicalDistributionOperator node, ExpressionContext context) {
                case BROADCAST:
                    if (statistics.getOutputSize(outputColumns) > sessionVariable.getMaxExecMemByte()) {
                        return CostEstimate.infinite();
                    }
                    int parallelExecInstanceNum = Math.max(1, getParallelExecInstanceNum(context));
                    // beNum is the number of right table should broadcast, now use alive backends
                    int beNum = Math.max(1, Catalog.getCurrentSystemInfo().getBackendIds(true).size());
                    result = CostEstimate
                            .of(statistics.getOutputSize(outputColumns) *
                                            Catalog.getCurrentSystemInfo().getBackendIds(true).size(),
                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum,
                                    statistics.getOutputSize(outputColumns) * beNum * parallelExecInstanceNum);
                    break;

...


        private int getParallelExecInstanceNum(ExpressionContext context) {
            return Math.min(ConnectContext.get().getSessionVariable().getDegreeOfParallelism(),
                    context.getRootProperty().getLeftMostScanTabletsNum());
        }
```

Reply:
@imay @kangkaisen PTAL